torch/csrc/jit/serialization/python_print.cpp

```diff
@@ -795,12 +800,47 @@ struct PythonPrintImpl {
     }
   }
 
+  static bool ContainsNonASCIIString(const IValue& val) {
```

Comment:
The implementation seems reasonable, but feels likely to be fragile over time. For example, if we add a new constant-able container type (`set` for example) and someone forgets to update this snippet of code, it will be a bug.

I've been seeing a lot of code that looks like this lately (deepcopy is another example)—I think we ought to implement a structure traversal iterator on IValue, so there's only one place we have to do this.

Reply:
also nit: no capital at the beginning (i.e. should be `containsNonAsciiString`)